### PR TITLE
nvidia-docker 1.0.1 (new formula)

### PR DESCRIPTION
**Pull request by cybertk**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/nvidia-docker.rb [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+class NvidiaDocker < Formula
+  desc "Build and run Docker containers leveraging NVIDIA GPUs"
+  homepage "https://github.com/NVIDIA/nvidia-docker"
+  url "https://github.com/NVIDIA/nvidia-docker/archive/v1.0.1.tar.gz"
+  sha256 "bb7cf057fadc5935c3ea2f7320516cae8feb8fc2421af9772e3127b258e38acd"
+
+  depends_on "go" => :build
+
+  patch :p1, :DATA
```
Comment (bfontaine) [inline Formula/nvidia-docker.rb]: Please report this bug to upstream

File: Formula/nvidia-docker.rb [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        :revision => "adbae1b6b6fb4b02448a0fc0dbbc9ba2b95b294d"
+  end
+
+  def install
+    ENV["GOPATH"] = buildpath/"go"
+
+    (buildpath/"go/src/github.com/nvidia/nvidia-docker").install buildpath.children
+    Language::Go.stage_deps resources, buildpath/"go/src"
+
+    cd("go/src/github.com/nvidia/nvidia-docker/src/nvidia-docker") do
```
Comment (bfontaine) [inline Formula/nvidia-docker.rb]: Nitpick: you can drop the parentheses.

File: Formula/nvidia-docker.rb [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+      system "go", "build", "-o", bin/"nvidia-docker"
+    end
+  end
+
+  test do
+    system "#{bin}/nvidia-docker", "version"
```
Comment (bfontaine) [inline Formula/nvidia-docker.rb]: Can’t we write a test that does something more substantial than e.g. `help`  or `version`?

**Review by bfontaine — CHANGES_REQUESTED** [review]

File: Formula/nvidia-docker.rb (hunk 1 above)
Comment (ilovezfs) [inline Formula/nvidia-docker.rb]: The patch will need to be accepted upstream before this can be merged.

Comment (cybertk) [inline Formula/nvidia-docker.rb]: This is not a bug, it's a customization for supporting macOS.

File: Formula/nvidia-docker.rb (hunk 3 above)
Comment (cybertk) [inline Formula/nvidia-docker.rb]: That requires a workable docker-engine, how to prepare this in test?

File: Formula/nvidia-docker.rb [hunk 4]
```diff
@@ -0,0 +1,55 @@
+require "language/go"
+require File.expand_path("../Requirements/cuda_requirement", __FILE__)
```
Comment (ilovezfs) [inline Formula/nvidia-docker.rb]: What is this referring to? I don't think we have any such requirement.

File: Formula/nvidia-docker.rb [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+class NvidiaDocker < Formula
+  desc "Build and run Docker containers leveraging NVIDIA GPUs"
+  homepage "https://github.com/NVIDIA/nvidia-docker"
+  url "https://github.com/NVIDIA/nvidia-docker/archive/v1.0.1.tar.gz"
+  sha256 "bb7cf057fadc5935c3ea2f7320516cae8feb8fc2421af9772e3127b258e38acd"
+
+  depends_on "go" => :build
+  depends_on CudaRequirement => :optional
+
+  patch :p1, :DATA
```
Comment (ilovezfs) [inline Formula/nvidia-docker.rb]: New formulae need to build without patching. Has this been reported upstream?

**Review by ilovezfs — CHANGES_REQUESTED** [review]

**Comment by fxcoudert** [comment]
Given lack of feedback, and despite the bots fighting each other… I am closing this for now. Please open a new pull request when the feedback is taken into account, and the formula builds.

**Pull request closed without merging**